### PR TITLE
refactor(calcite-dropdown-item): stop spreading attributes

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #1579

### CODE REVIEW

File: src/components/calcite-dropdown-item/calcite-dropdown-item.tsx [hunk 1]
```diff
@@ -142,7 +133,14 @@ export class CalciteDropdownItem {
...
       slottedContent
     ) : (
-      <a {...attributes} class="dropdown-link" ref={(el) => (this.childLink = el)}>
+      <a
+        class="dropdown-link"
+        href={this.href}
+        ref={(el) => (this.childLink = el)}
+        rel={this.el.getAttribute("rel")}
+        target={this.el.getAttribute("target")}
+        title={this.el.getAttribute("title")}
```
Comment (eriklharper) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.tsx:142]: @driskull should we make explicit props for these as well?

Comment (driskull) [reply src/components/calcite-dropdown-item/calcite-dropdown-item.tsx:142]: Yes 👍 

For props that already exist at global level like 'title' we will need to use label or something.

Comment (driskull) [reply src/components/calcite-dropdown-item/calcite-dropdown-item.tsx:142]: explicit props means that they are documented and supported. Unlike before :)

Comment (eriklharper) [reply src/components/calcite-dropdown-item/calcite-dropdown-item.tsx]: I matched what we just did with `calcite-button` and created a label prop to apply to `aria-label` on the `<a>` and ditch setting the `title` attribute.

**Review by driskull — APPROVED** [review]
👍 Sounds good. If we need a way to set a title (tooltip) we can do that later.

**Pull request merged**